StRoot/StEvent/StEnumerations.cxx

```diff
@@ -3,10 +3,64 @@
 #include "TROOT.h"
 #include "TSystem.h"
 #include "TString.h"
-int         ids[100]={0};
-const char *cds[100]={0};
+int   ids[kMaxDetectorId+2]={0};
+char *cds[kMaxDetectorId+2]={0};
 static void detectorIdInit();
 //_____________________________________________________________________________
+void detectorId(int *ids=0, char** cds=0)
+{
+ int   myIds[kMaxDetectorId+2];
+ char *myCds[kMaxDetectorId+2];
+ if (!ids) { ids = myIds; cds = myCds; }
+
+ memset(ids,0,sizeof(ids[0])*kMaxDetectorId+2);
+ memset(cds,0,sizeof(cds[0])*kMaxDetectorId+2);
+
+ TString myPath("$STAR/StRoot/StEvent/StEnumerations.h");
+ gSystem->ExpandPathName(myPath);
+
+ int notExi = gSystem->AccessPathName(myPath.Data(),kFileExists);
+ if (notExi)    { ids[0]=-1; return;}
+ FILE *fp = fopen(myPath.Data(),"r");
+ if (!fp)       { ids[0]=-1; return;}
+ char buf[400];
+
+ int kase = 0;
+ while (2015) {
```

Comment:
Why `2015`?

Reply:
Possibly b/c Victor added this code in 2015?  Regardless, the point was to have an infinite loop, which is broken out of when the termination of the StDetectorId enum block was found.